lib/fencing/st_client.c

```diff
@@ -718,11 +717,8 @@ stonith_api_history(stonith_t * stonith, int call_options, const char *node,
             crm_element_value_ll(op, F_STONITH_DATE_NSEC, &completed_nsec);
             kvp->completed_nsec = completed_nsec;
             crm_element_value_int(op, F_STONITH_STATE, &kvp->state);
-
-            stonith__xe_get_result(op, &result);
-            kvp->exit_reason = result.exit_reason;
-            result.exit_reason = NULL;
-            pcmk__reset_result(&result);
+            kvp->exit_reason = crm_element_value_copy(op,
+                                                      XML_LRM_ATTR_EXIT_REASON);
```

Comment:
This looks good to me. The previous approach seems like overkill anyway despite looking clean.